userauth.go

```diff
@@ -160,20 +160,22 @@ func (u *ValidateFromFile) secretFromID(id string) (string, error) {
 
 // Authenticate verifies that the token included in the http.Request
 // is valid
-func (u *ValidateFromToken) Authenticate(r *http.Request) error {
+func (u *ValidateFromToken) Authenticate(r *http.Request) (claims jwt.MapClaims, err error) {
+	var ok bool
+
 	// Verify signature by parsing the token with the given key
 	tokenStr := r.Header.Get("X-Amz-Security-Token")
 	if tokenStr == "" {
-		return fmt.Errorf("no access token supplied")
+		return nil, fmt.Errorf("no access token supplied")
 	}
 
 	token, err := jwt.Parse(tokenStr, func(tokenStr *jwt.Token) (interface{}, error) { return nil, nil })
 	// Return error if token is broken (without claims)
-	if claims, ok := token.Claims.(jwt.MapClaims); !ok {
-		return fmt.Errorf("broken token (claims are empty): %v\nerror: %s", claims, err)
+	if claims, ok = token.Claims.(jwt.MapClaims); !ok {
+		return nil, fmt.Errorf("broken token (claims are empty): %v\nerror: %s", claims, err)
 	}
 	//nolint:nestif
-	if claims, ok := token.Claims.(jwt.MapClaims); ok {
+	if claims, ok = token.Claims.(jwt.MapClaims); ok {
```

Comment:
This should be refactored as the previous if statement will block access to here if it fails.
In short this if statement is fully redundant and provide no real value.

Reply:
Good catch, didn't notice it

Reply:
With the previous implementation it was a different variable between the if statements, but this is no longer the case, so I am removing it